File: ege/07/17.py
# 17
numbers = [int(i) for i in open('/home/ellion/PycharmProjects/pythonProject/ege/07/17.txt').readlines()]
s = sum(numbers) / len(numbers)
count, maxsum = 0, 0
for i in range(len(numbers)-2):
    a = numbers[i]
    b = numbers[i+1]
    c = numbers[i+2]
    isdel = [a % 7 == 0, b % 7 == 0, c % 7 == 0]
    if (min(a, b, c) < s) and isdel.count(True) >= 2:
        count += 1
        maxsum = max(maxsum, a + b + c)
print(count, maxsum)

File: ege/07/17.go
package main

import (
	"bufio"
	"os"
	"strconv"
)

func main() {
	file, err := os.Open("/home/ellion/ege/07/17.txt")
	if err != nil {
		panic(err)
	}
	var maxsum int16 = 0
	count := 0
	x := 0
	numbers := [10000]int16{}
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		lineStr := scanner.Text()
		num, _ := strconv.Atoi(lineStr)
		numbers[x] = int16(num)
		x++
	}
	defer file.Close()
	for index1, item1 := range numbers[:9998] {

		if maxsum < item1+item1 {
			maxsum = item1 + int16(index1)

		}
	}
	println(count, maxsum)
}
